### PR TITLE
Drop support for Python < 3.4

### LINKED ISSUE #259: Drop Python 2 support
As Django 2.0 dropped it, it might be a good time for APScheduler to drop Python 2 support too. This would open the possibility of adding native support for typing information directly into the source code, and to remove some awkward compatibility code. Python 2 users could still keep using 3.x, which should get bug fixes when possible but no new features.

### PR DESCRIPTION
**Pull request by hugovk**
Fixes #259.

### CODE REVIEW

**Comment by dpgaspar** [comment]
please don't...

**Comment by agronholm** [comment]
Please don't what?

**Comment by agronholm** [comment]
This PR won't be merged for a long while – 4.0 is still waaay ahead in the future.

File: apscheduler/jobstores/mongodb.py [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
 
-from apscheduler.jobstores.base import BaseJobStore, JobLookupError, ConflictingIdError
-from apscheduler.util import maybe_ref, datetime_to_utc_timestamp, utc_timestamp_to_datetime
 from apscheduler.job import Job
-
-try:
-    import cPickle as pickle
-except ImportError:  # pragma: nocover
-    import pickle
+from apscheduler.jobstores.base import (BaseJobStore, ConflictingIdError,
```
Comment (agronholm) [inline apscheduler/jobstores/mongodb.py]: Getting rid of the conditional pickle import was fine, but why did you rearrange these other imports? They were fine as they were.

File: apscheduler/jobstores/mongodb.py [hunk 2]
```diff
@@ -1,14 +1,13 @@
 from __future__ import absolute_import
```
Comment (agronholm) [inline apscheduler/jobstores/mongodb.py]: All `__future__` imports should be removed too.

File: apscheduler/jobstores/zookeeper.py [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
-
-try:
-    import cPickle as pickle
-except ImportError:  # pragma: nocover
-    import pickle
+from apscheduler.jobstores.base import (BaseJobStore, ConflictingIdError,
+                                        JobLookupError)
+from apscheduler.util import (datetime_to_utc_timestamp, maybe_ref,
+                              utc_timestamp_to_datetime)
+from kazoo.exceptions import NodeExistsError, NoNodeError
```
Comment (agronholm) [inline apscheduler/jobstores/zookeeper.py]: Exception in wrong group.

File: apscheduler/jobstores/mongodb.py (hunk 1 above)
Comment (hugovk) [reply apscheduler/jobstores/mongodb.py]: I used isort https://github.com/timothycrosley/isort
Shall I unsort?

File: apscheduler/jobstores/zookeeper.py (hunk 3 above)
Comment (hugovk) [reply apscheduler/jobstores/zookeeper.py]: Same thing. Shall I unsort?

File: apscheduler/jobstores/mongodb.py (hunk 2 above)
Comment (hugovk) [reply apscheduler/jobstores/mongodb.py]: Removed.

File: apscheduler/jobstores/mongodb.py (hunk 1 above)
Comment (agronholm) [reply apscheduler/jobstores/mongodb.py]: Yes, the sort apparently folds import lines unnecessarily.

Comment (hugovk) [reply apscheduler/jobstores/mongodb.py]: Unsorted this and the other.

Comment (agronholm) [reply apscheduler/jobstores/mongodb.py]: There are plenty more imports still in this PR where two import lines were changed into two.

Comment (hugovk) [reply apscheduler/jobstores/mongodb.py]: Unsorted!

**Comment by agronholm** [comment]
Dude, you don't need to do this. I have this handled.

**Comment by hugovk** [comment]
Sure thing, just resolving merge conflicts.

**Pull request closed without merging**